tools/ci_build/github/azure-pipelines/win-ci-pipeline.yml

```diff
@@ -4,39 +4,103 @@ parameters:
+  jobs:
+    - template: templates/win-cpu-ci.yml
+      parameters:
+        BuildConfig: 'RelWithDebInfo'
+        UseOmp: ''
+        EnvSetupScript: setup_env.bat
+        buildArch: x64
+        # Compare to our Nuget packaging pipeline, this job has "--build_wheel --use_dnnl --build_nodejs" but doesn't have "--enable_lto --disable_rtti --use_telemetry  --enable_wcos"
+        # Python bindings use typeid so I can't disable RTTI here. So we may need to split it to two pipelines
+        additionalBuildFlags: --use_dnnl --build_java --build_nodejs
+        msbuildPlatform: x64
+        isX86: false
+        job_name_suffix: x64_release
+        RunOnnxRuntimeTests: ${{ parameters.RunOnnxRuntimeTests }}
+        RunStaticCodeAnalysis: true
 
-- template: templates/win-cpu-ci.yml
-  parameters:
-    BuildConfig: 'RelWithDebInfo'
-    UseOmp: ''
-    EnvSetupScript: setup_env_x86.bat
-    buildArch: x86
-    additionalBuildFlags:
-    msbuildPlatform: Win32
-    isX86: true
-    job_name_suffix: x86_release
-    RunOnnxRuntimeTests: ${{ parameters.RunOnnxRuntimeTests }}
\ No newline at end of file
+- stage: x64_release_winml
+  dependsOn: []
+  jobs:
+    - template: templates/win-cpu-ci.yml
+      parameters:
+        BuildConfig: 'RelWithDebInfo'
+        UseOmp: ''
+        EnvSetupScript: setup_env.bat
+        buildArch: x64
+        additionalBuildFlags: --use_winml --enable_wcos --disable_rtti
+        msbuildPlatform: x64
+        isX86: false
+        job_name_suffix: x64_release_winml
+        RunOnnxRuntimeTests: ${{ parameters.RunOnnxRuntimeTests }}
+        # WinML has many warnings
+        RunStaticCodeAnalysis: false
+
+- stage: x86_release
+  dependsOn: []
+  jobs:
+    - template: templates/win-cpu-ci.yml
+      parameters:
+        BuildConfig: 'RelWithDebInfo'
+        UseOmp: ''
+        EnvSetupScript: setup_env_x86.bat
+        buildArch: x86
+        additionalBuildFlags: 
+        msbuildPlatform: Win32
+        isX86: true
+        job_name_suffix: x86_release
+        RunOnnxRuntimeTests: ${{ parameters.RunOnnxRuntimeTests }}
+        RunStaticCodeAnalysis: true
+
+- stage: training_x64_debug
+  dependsOn: []
+  jobs:
+    - template: templates/win-cpu-ci.yml
+      parameters:
+        BuildConfig: 'Debug'
+        UseOmp: ''
+        EnvSetupScript: setup_env.bat
+        buildArch: x64
+        additionalBuildFlags: --enable_training
+        msbuildPlatform: x64
+        isX86: false
+        job_name_suffix: training_x64_debug
+        RunOnnxRuntimeTests: ${{ parameters.RunOnnxRuntimeTests }}
+        RunStaticCodeAnalysis: false
+
+- stage: training_x64_release
+  dependsOn: []
+  jobs:
+    - template: templates/win-cpu-ci.yml
+      parameters:
+        BuildConfig: 'RelWithDebInfo'
+        UseOmp: ''
+        EnvSetupScript: setup_env.bat
+        buildArch: x64
+        additionalBuildFlags: --enable_training
+        msbuildPlatform: x64
+        isX86: false
+        job_name_suffix: training_x64_release
```

Comment:
does this work?
`${{ variables['System.StageName'] }}`

Reply:
No, it doesn't.